scripts/configurator/parser/cmake_parser.py

```diff
@@ -0,0 +1,37 @@
+from cmake_row import cmake_row, configurator_type, row_type
+from typing import List
+
+def read_file(filepath):
+   with open(filepath) as f:
+      return f.readlines()
+
+def lines_to_rows(file_lines: List[str]) -> List[cmake_row]:
+   """
+   turns the lines of a file (parsed previously) to a list of cmake_rows,
+   skipping internal variables.
+   """
+   cmake_rows: List[cmake_row] =  []
+   comment = ""
+
+   for i, line in enumerate(file_lines):
+      row = cmake_row(line, i, comment)
+      if row.row_type == row_type.SLASH_COMMENT:
+         comment += row.val
+      elif row.row_type == row_type.VARIABLE and \
+            row.configurator_type != configurator_type.INTERNAL:
+         cmake_rows.append(row)
```

Comment:
1. why are you appending to rows only the variables? I believe we talked about this: in this approach ALL the lines got converted to cmake_row instances.

2. you're appending too many comments to variables. This approach is not OK. Either you walk back the comments once if found a variable or you walk always forward but reset the comment when you hit an empty line. Btw, the second approach is less intuitive to both read & write.